ios/RCTHTTPRequestHandler+AuthenticationChallenge.m

```diff
@@ -0,0 +1,189 @@
+                                      task:(NSURLSessionTask *)task
+                       didReceiveChallenge:(NSURLAuthenticationChallenge *)challenge
+                         completionHandler:(void (^)(NSURLSessionAuthChallengeDisposition disposition, NSURLCredential *credential))completionHandler {
+  // Set of allowed domains
+  NSSet *allowedDomains = [self getAllowedDomains];
+  // Check if the challenge is of type ServerTrust
+  if ([challenge.protectionSpace.authenticationMethod isEqualToString:NSURLAuthenticationMethodServerTrust]) {
+    // Check if the host is an allowed domain
+    if ([self checkValidityOfHost:challenge.protectionSpace.host allowedInDomains:allowedDomains]) {
+      // Check if proceed with the original authentication handling
+      if ([self checkValidityOfTrust:challenge.protectionSpace.serverTrust]) {
+        // The challenge is of type ServerTrust, the host is allowed and the certificate is trust worthy
+        // Create the credential and completes the delegate
+        NSURLCredential *credential = [NSURLCredential credentialForTrust:challenge.protectionSpace.serverTrust];
+        completionHandler(NSURLSessionAuthChallengeUseCredential, credential);
+      } else {
+        // Certificate is invalid, cancelling authentication
+        NSLog(@"Invalid certificate for Domain %@. Cancelling authentication.", challenge.protectionSpace.host);
+        // Cancel the authentication challenge
+        completionHandler(NSURLSessionAuthChallengeCancelAuthenticationChallenge, nil);
+      }
+    } else {
+      // Host is not allowed, cancelling authentication
+      NSLog(@"Host domain %@ is not allowed. Cancelling authentication.", challenge.protectionSpace.host);
+      // Cancel the authentication challenge
+      completionHandler(NSURLSessionAuthChallengeCancelAuthenticationChallenge, nil);
+    }
+  } else {
+    // Trust is not of type ServerTrust, proceeding with default challenge, if any
+    NSLog(@"Not ServerTrust challenge. Calling default authentication challenge.");
+    // For other authentication methods, call the original implementation
+    [self authenticationChallenge_URLSession:session task:task didReceiveChallenge:challenge completionHandler:completionHandler];
+  }
+}
+
+- (BOOL)checkValidityOfTrust:(SecTrustRef)serverTrust {
+  CFErrorRef error = NULL;
+  BOOL isValid = SecTrustEvaluateWithError(serverTrust, &error);
+  
+  if (!isValid) {
+    /* This is a casting a core foundation class, CFErrorRef, to an Objective-C class, NSError.
+     * It means the new error instance will be managed by Objective-C, which will control the
+     * memory for this object, making our work easier.
+     */
+    NSError *nsError = (__bridge_transfer NSError *)error;
+    NSLog(@"Trust evaluation failed: %@", nsError);
+  }
+  
+  return isValid;
+}
+
+- (NSSet *)getAllowedDomains {
+  NSDictionary *infoDictionary = [[NSBundle mainBundle] infoDictionary];
```

Comment:
Should we also throw an error if `NSAppTransportSecurity` or `NSPinnedDomains` are not found in `Info.plist`?

Reply:
Oh! I think so. I'm going to add this.

Reply:
Done at: e447699556130341b2c10bd3d9b32c90416f34c3